src/components/PopupWithForm.js

```diff
@@ -26,4 +29,12 @@ export default class PopupWithForm extends Popup {
     });
     super.setEventListeners();
   }
+
+  showButtonProgress(showButtonProgress) {
```

Comment:
You've implemented the code for rendering loading via the class methods.Great job!